### PR TITLE
Add template methods

### PR DESCRIPTION
**Pull request by servingUpAces**
* Added methods to get templates and generate a preview of a template.
* `get_template_by_id` - get the latest version of a template by id.
* `get_template_version` - get the template by id and version.
* `get_all_templates` - get all templates, can be filtered by template type.
* `generate_template_preview` - get the contents of a template with the placeholders replaced with the given personalisation.

### CODE REVIEW

File: bin/test_client.rb [hunk 1]
```diff
@@ -99,6 +173,22 @@ def field_should_be_nil(fields, obj, method_name)
...
 end
 
+def expected_fields_in_template_response
+  %w(id
+     type
+     created_at
+     created_by
+     body
+     version
+)
```
Comment (imdadahad) [inline bin/test_client.rb]: `updated_at` and `type`?

File: bin/test_client.rb [hunk 2]
```diff
@@ -99,6 +173,22 @@ def field_should_be_nil(fields, obj, method_name)
...
+end
+
+def expected_fields_in_template_preview
+  %w(id
+     body
+     version
+)
```
Comment (imdadahad) [inline bin/test_client.rb]: `type`?

File: bin/test_client.rb [hunk 3]
```diff
@@ -99,6 +173,22 @@ def field_should_be_nil(fields, obj, method_name)
...
+end
```
Comment (imdadahad) [inline bin/test_client.rb]: Also do we have a check to see that the `subject` exists for an `email` preview?

File: lib/notifications/client/response_template.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            ].each do |field|
+              define_method field do
+                begin
+                  value = instance_variable_get(:"@#{field}")
+                  Time.parse value
+                rescue
+                  value
+                end
+              end
+            end
```
Comment (imdadahad) [inline lib/notifications/client/response_template.rb]: This is hard to read :-|, might just be Ruby...

File: spec/factories/template_preview.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    initialize_with do
+      new(body)
+    end
+
+    body do
+      {
+        "id" => "f163deaf-2d3f-4ec6-98fc-f23fa511518f",
+        "body" => "Contents of template Mr Big Nose",
+        "subject" => "Subject of the email",
+        "version" => "2"
```
Comment (imdadahad) [inline spec/factories/template_preview.rb]: This should match what we expect (refer to comment above).

File: bin/test_client.rb (hunk 1 above)
Comment (servingUpAces) [inline bin/test_client.rb]: type is there and updated_at can be null

**Pull request merged**